frontend/hooks/checkWriteable.ts

```diff
@@ -0,0 +1,71 @@
+import { useAccount, useBalance, useNetwork } from "wagmi";
+import { toast } from "react-toastify";
+import { DEFAULT_CHAIN_ID } from "../lib/config";
+import { useHypercertClient } from "./hypercerts-client";
+import { useEffect, useState } from "react";
+
+const useCheckWriteable = () => {
+  const { address, isConnected } = useAccount();
+  const { chain } = useNetwork();
+  const { client } = useHypercertClient();
+  const { data: balance } = useBalance({
+    address,
+    enabled: !!address,
+  });
+  const [writeable, setWriteable] = useState(false);
+
+  useEffect(() => {
+    const check = async () => {
+      const writeable = await checkWriteable(true);
+      setWriteable(writeable);
+    };
+    check();
+  }, [address, isConnected, balance, chain, client]);
+
+  const checkWriteable = async (silent = false) => {
+    if (!address || !isConnected) {
+      console.log("User not connected");
+      !silent ? toast("Please connect your wallet", { type: "error" }) : null;
```

Comment:
I feel like it'd be cleaner to store the errors in a `useState` that gets returned with `writeable` from the hook. That way the caller can decide how to present the error themselves, whether it's a toast or something else?

What do you think? That way we don't need a `silent` option, the caller can just deal with the error.

Reply:
Ah yes. Think I took the short way home here. Will refactor this, that should also resolve the other comments